daemon/endpoint.go

```diff
@@ -211,6 +211,44 @@ func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder
 	if err != nil {
 		apierror.Error(code, err)
 	}
+
+	// Wait for endpoint to be ready if specified in API call.
+	if params.Endpoint.SyncBuildEndpoint {
+		log.Debug("synchronously waiting for endpoint to be in %s state", endpoint.StateReady)
```

Comment:
We should include at least an endpoint ID in the log message here

Reply:
Done.